geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/STIdxStrategy.scala

```diff
@@ -53,18 +54,18 @@ class STIdxStrategy(val filter: QueryFilter) extends Strategy with LazyLogging w
     output(s"Temporal filters: ${filtersToString(temporalFilters)}")
 
     // standardize the two key query arguments:  polygon and date-range
-    val geomsToCover = geomFilters.flatMap(decomposeToGeometry)
 
-    output(s"GeomsToCover: $geomsToCover")
+    val geometryToCover =
+      andOption(geomFilters).flatMap(extractSingleGeometry(_, sft.getGeomField)).getOrElse(WholeWorldPolygon)
 
-    val collectionToCover: Geometry = geomsToCover match {
-      case Nil => null
-      case seq: Seq[Geometry] => new GeometryCollection(geomsToCover.toArray, geomsToCover.head.getFactory)
+    val interval = {
+      val intervals = for { dtg <- dtgField; filter <- andOption(temporalFilters) } yield {
+        extractIntervals(filter, dtg)
+      }
+      // note: because our filters were and'ed, there will be at most one interval
+      intervals.flatMap(_.headOption).map(i => new Interval(i._1, i._2)).orNull
```

Comment:
This might be cleaner as a match statement.  

Maybe extractIntervals should return an Option?